src/OpenTelemetry.Exporter.Geneva/TLDExporter/TldTraceExporter.cs

```diff
@@ -117,29 +114,29 @@ public TldTraceExporter(GenevaExporterOptions options)
 
     public ExportResult Export(in Batch<Activity> batch)
     {
-        var result = ExportResult.Success;
         if (this.eventProvider.IsEnabled())
         {
+            var result = ExportResult.Success;
+
             foreach (var activity in batch)
             {
                 try
                 {
-                    this.SerializeActivity(activity);
-                    this.eventProvider.Write(EventBuilder.Value);
+                    var eventBuilder = this.SerializeActivity(activity);
+
+                    this.eventProvider.Write(eventBuilder);
                 }
                 catch (Exception ex)
                 {
                     ExporterEventSource.Log.FailedToSendTraceData(ex); // TODO: preallocate exception or no exception
                     result = ExportResult.Failure;
                 }
             }
-        }
-        else
-        {
-            return ExportResult.Failure;
+
+            return result;
         }
 
-        return result;
+        return ExportResult.Failure;
```

Comment:
Looks like we may have been incorrectly returning `Success` when the eventProvider was not enabled.
Does this need a bug fix line in the Changelog?

Reply:
I think it was fine before just kind of awkwardly written. All I did here was switch it around to hopefully make it more readable.